DummyFile.java

```diff
@@ -1,4 +1,40 @@
+import com.amazonaws.services.kinesis.AmazonKinesisClient;
+import com.amazonaws.services.kinesis.AmazonKinesisClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.InitiateMultipartUploadRequest;
+import com.amazonaws.services.s3.model.ListObjectsRequest;
+import com.amazonaws.services.s3.model.ListObjectsV2Request;
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+
+import example.foo.FooClient;
+
 public class DummyFile {
-    jkgfldksghkjfdhgkfdgkfdhsjnvjkfdnvkjfdnvkjf
-        gfdgdfjkgkdj
+    private AmazonS3 s3Client = AmazonS3ClientBuilder.standard().build();
+    private AmazonKinesisClient kinesisClient = AmazonKinesisClientBuilder.defaultClient();
+    private FooClient fooClient;
+
+    // Should flag and suggest v2
+    public void s3ListObjects() {
+        s3Client.listObjects(new ListObjectsRequest());
+    }
+
+    // Should not flag
+    public void s3ListObjectsV2() {
+        s3Client.listObjectsV2(new ListObjectsV2Request());
+    }
+
+    // Should flag and suggest Transfer Manager
+    public void s3MultiPartUpload() {
+        s3Client.initiateMultipartUpload(new InitiateMultipartUploadRequest());
+    }
+
+    // Should not flag
+    public void wrongReceiver() {
+        fooClient.listObjects(new ListObjectsRequest());
+    }
+
+    // Should flag
+    public void kinesisEnhancedFanOut() {
+        kinesisClient.getRecords(null);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Consider enhanced fan-out: an Amazon Kinesis Data Streams feature that enables consumers to receive records from a data stream with dedicated throughput of up to 2 MiB of data per second per shard. A consumer that uses enhanced fan-out doesn&#39;t have to contend with other consumers that are receiving data from the stream. See [Developing Consumers with Enhanced Fan-Out Using the Kinesis Data Streams API](https://docs.aws.amazon.com/streams/latest/dev/building-enhanced-consumers-api.html).